browser-extension/mv3/src/service-worker/services/rulesManager.ts

```diff
@@ -104,11 +107,29 @@ const addExtensionRules = async (): Promise<void> => {
 };
 
 const applyExtensionRules = async (): Promise<void> => {
-  await deleteAllDynamicRules();
+  ruleApplicationQueue.push(async () => {
+    try {
+      await deleteAllDynamicRules();
+      const isExtensionStatusEnabled = await isExtensionEnabled();
+      if (isExtensionStatusEnabled) {
+        await addExtensionRules();
+      }
+    } finally {
+      isApplyingRules = false;
+      processRuleApplicationQueue();
+    }
+  });
+
+  if (!isApplyingRules) {
+    processRuleApplicationQueue();
+  }
+};
```

Comment:
Since we remove all the DNR rules and then find the enabled rules everytime. So the end results always remains the same. So instead of making a queue, should we add debounce instead. This will make sure all the DNR aren't sent again and again to chrome (possible performance issues when setting DNRs again and again?)

Reply:
1 downside i see is that when we perform bulk operation on large number of rules, rules won't  work unless all the rules `applyDynamicRules` listeners are processed and then DNRs are applied. But its the same as queue as in that case too all the rules will be removed on every `applyDynamicRules`

Reply:
Implemented debounce